pyjobshop/solvers/ortools/Constraints.py

```diff
@@ -271,30 +270,23 @@ def _circuit_constraints(self):
                 continue
 
             modes = seq_var.mode_vars
-            starts = seq_var.starts
-            ends = seq_var.ends
-            ranks = seq_var.ranks
             arcs = seq_var.arcs
 
             # Add dummy node self-arc to allow empty circuits.
             empty = model.new_bool_var(f"empty_circuit_{idx}")
```

Comment:
The simplification that you introduced is very helpful to cut down the complexity of sequence variables. I think from a coding point-of-view we can even simplify this further.

- Create all variables in `SequenceVar.activate`
- Only keep constraints in `circuit_constraints`

I think this separation is clearer. Do you want to give it a shot?

Reply:
On a second thought, I don't think it matters too much. Let's not do this.